src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -2144,13 +2144,9 @@ - (NSArray *)_getAttributesToReportWithReportedValues:(NSArray<NSDictionary<NSSt
             // if expected values exists, purge and update read cache
             NSArray * expectedValue = _expectedValueCache[attributePath];
             if (expectedValue) {
-                if (![self _attributeDataValue:attributeDataValue
-                            isEqualToDataValue:expectedValue[MTRDeviceExpectedValueFieldValueIndex]]) {
-                    shouldReportAttribute = YES;
-                    previousValue = expectedValue[MTRDeviceExpectedValueFieldValueIndex];
-                }
-                _expectedValueCache[attributePath] = nil;
+                previousValue = expectedValue[MTRDeviceExpectedValueFieldValueIndex];
```

Comment:
Why are we setting `previousValue` to anything?  It should only be set (and is only used) when shouldReportAttribute == YES.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/33013